pkg/registry/core/service/allocator/bitmap.go

```diff
@@ -205,3 +254,37 @@ func (rss randomScanStrategy) AllocateBit(allocated *big.Int, max, count int) (i
 }
 
 var _ bitAllocator = randomScanStrategy{}
+
+// randomScanReservedStrategy choose a random address from the provided big.Int and then scans
+// forward looking for the next available address. The big.Int range is subdivided so it will try
+// to allocate first from the reserved upper range of addresses (it will wrap the upper subrange if necessary).
+// If there is no free address it will try to allocate one from the lower range too.
+type randomScanReservedStrategy struct {
+	rand     *rand.Rand
+	reserved int
+}
+
+func (rss randomScanReservedStrategy) AllocateBit(allocated *big.Int, max, count int) (int, bool) {
+	if count >= max {
+		return 0, false
+	}
+	// size of the upper subrange, prioritized for random allocation
+	subrangeMax := max - rss.reserved
+	// try to get a value from the upper range [rss.reserved, max]
+	offset := rss.rand.Intn(subrangeMax)
+	for i := 0; i < subrangeMax; i++ {
+		at := rss.reserved + ((offset + i) % subrangeMax)
+		if allocated.Bit(at) == 0 {
+			return at, true
+		}
+	}
+	// subrange full, try to get the value from the first block before giving up.
+	for i := 0; i < rss.reserved; i++ {
```

Comment:
Shouldn't we random-seed this too?

Reply:
> 	// default $min = 16, $max = 256 and $step = 16.

I thought about that but since we are capping it to 256 max I didn't want to complicate  this loop because there is no  much performance improvement

Reply:
Maybe do this in reverse-order, then?  It seems like the lower numbers are more likely to be static?